omod/src/main/webapp/portlets/flagPatientOverview.jsp

```diff
@@ -1,8 +1,22 @@
+<%@page import="org.openmrs.module.patientflags.Flag"%>
 <%@ include file="/WEB-INF/template/include.jsp"%>
 
 <openmrs:require privilege="View Patients" otherwise="/login.htm"
 	redirect="/index.htm" />
 
 <c:forEach items="${model.flags}" var="flag">
-	<span ${flag.priority != null ? flag.priority.style : ''}>${flag.localizedMessage}</span><br />
+	<span ${flag.priority.style}> 
+	<c:set var="flag" value="${flag}"/>
+	<c:set var="patient_id" value="${param.patientId}"/>
+	 <%
+	 try{
+	 Flag flag = (Flag)pageContext.getAttribute("flag");
+	 Integer patient_id = Integer.parseInt(pageContext.getAttribute("patient_id").toString());
+	 out.println(flag.evalMessage(patient_id));
+	 }
+	 catch(Exception e){
+		 e.printStackTrace();
+	 }
+	  %>

```

Comment:
I'd *much* prefer we keep the raw Java out of the JSP, and rather formulate the message in the controller and only use JSTL here.